source/MaterialXCore/Element.cpp

```diff
@@ -584,6 +586,26 @@ ValuePtr ValueElement::getDefaultValue() const
     return ValuePtr();
 }
 
+const string& ValueElement::getDefaultUnit() const
```

Comment:
What would you think of naming this method "getActiveUnit"?  To my mind, it's closely analogous to "getActiveColorspace", which applies any applicable rules (e.g. scoping, inheritance) to return the final colorspace that applies to the given element.

Reply:
This is fine with me. I think the suggestion for getDefaultUnit() was from you originally.